### PR TITLE
William/typescript

### PR DESCRIPTION
**Pull request by swansontec**
I have finished rebasing #3640, but it's so different now that I am submitting the work as a new PR.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -99,7 +99,7 @@
     "assert": "^2.0.0",
     "async-lock": "^1.1.2",
     "base-x": "^4.0.0",
-    "biggystring": "^3.0.2",
+    "biggystring": "^4.0.0",
```
Comment (paullinator) [inline package.json:99]: yarn.lock wasn't updated.

File: package.json [hunk 2]
```diff
@@ -187,6 +188,7 @@
     "@types/react-test-renderer": "^18.0.0",
     "@types/sprintf-js": "^1.1.2",
     "@types/url-parse": "^1.4.8",
+    "@types/uuid": "^7.0.0",
```
Comment (paullinator) [inline package.json:188]: no yarn.lock change

File: src/components/scenes/Loans/LoanStatusScene.js [hunk 3]
```diff
@@ -103,7 +103,7 @@ export const LoanStatusScene = (props: Props) => {
           ) : (
             <MainButton label={s.strings.loan_status_cancel_txs} type="secondary" onPress={handleCancelPress} marginRem={[0.5, 1, 2, 1]} />
           )}
-          {isProgramDone ? <ConfettiCannon autostart count={250} origin={{ x: -50, y: -50 }} fallSpeed={4000} /> : null}
+          {isProgramDone ? <ConfettiCannon autoStart count={250} origin={{ x: -50, y: -50 }} fallSpeed={4000} /> : null}
```
Comment (paullinator) [inline src/components/scenes/Loans/LoanStatusScene.js]: to be consistent with previous behavior, remove `autoStart` completely

File: src/modules/UI/components/Buttons/TertiaryButton.ui.js [hunk 4]
```diff
@@ -30,7 +30,7 @@ export class TertiaryButton extends React.Component<Props> {
   render() {
     const { children, style, ...props } = this.props
     return (
-      <TouchableHighlight underlayColor={rawStyles.tertiaryButtonUnderlay.color} style={[styles.button, styles.tertiaryButton, style]} {...props}>
+      <TouchableHighlight underlayColor={tertiaryButtonUnderlay.color} style={[styles.button, styles.tertiaryButton, style]} {...props}>
```
Comment (paullinator) [inline src/modules/UI/components/Buttons/TertiaryButton.ui.js]: does this work? you're passing a style parameter into a non-style prop. The previous code looks more correct

**Review by paullinator — CHANGES_REQUESTED** [review]
See comments

File: src/modules/UI/components/Buttons/TertiaryButton.ui.js (hunk 4 above)
Comment (swansontec) [reply src/modules/UI/components/Buttons/TertiaryButton.ui.js]: The `underlayColor` is a [normal prop](https://reactnative.dev/docs/0.68/touchablehighlight#underlaycolor), not a style for some reason. So we need to pass it as a normal value, which is really weird but correct.

File: src/components/scenes/Loans/LoanStatusScene.js (hunk 3 above)
Comment (swansontec) [reply src/components/scenes/Loans/LoanStatusScene.js]: Will do.

File: package.json (hunk 2 above)
Comment (swansontec) [reply package.json:188]: This version is already in yarn.lock, so no change needed.

File: package.json (hunk 1 above)
Comment (swansontec) [reply package.json:99]: This version is already in yarn.lock, so no change needed.

**Review by paullinator — APPROVED** [review]

File: src/modules/UI/components/Buttons/style.js [hunk 5]
```diff
@@ -5,7 +5,11 @@ import { StyleSheet } from 'react-native'
 import { THEME } from '../../../../theme/variables/airbitz'
 import { scale } from '../../../../util/scaling.js'
 
-export const rawStyles = {
+export const primaryButtonUnderlay = { color: THEME.COLORS.PRIMARY }
+export const secondaryButtonUnderlay = { color: THEME.COLORS.GRAY_1 }
+export const tertiaryButtonUnderlay = { color: THEME.COLORS.GRAY_3 }
+
+export const styles = StyleSheet.create({
```
Comment (swansontec) [inline src/modules/UI/components/Buttons/style.js]: @paullinator See how we have pulled the underlay exports out of the style block, and made them normal exports? This is why the files above work correctly.

**Pull request merged**